tests/unit/wallet/test_wallet.py

```diff
@@ -74,6 +82,62 @@ def test_load_and_save_wallet(self):
         decrypted = Wallet.unpack('password', encrypted)
         self.assertEqual(decrypted['accounts'][0]['name'], 'An Account')
 
+
+    def test_wallet_file_schema(self):
+        # One Deterministic Chain, one Single Address, to test both paths in the schema.
+        wallet_dict = {
+            'version': 1,
+            'name': 'Main Wallet',
+            'preferences': {},
+            'accounts': [
+                {
+                    'certificates': {'x': 'y'},
+                    'name': 'Account 1',
+                    'ledger': 'lbc_mainnet',
+                    'modified_on': 123,
+                    'seed':
+                        "carbon smart garage balance margin twelve chest sword toast envelope bottom stomac"
+                        "h absent",
+                    'encrypted': False,
+                    'private_key':
+                        'xprv9s21ZrQH143K42ovpZygnjfHdAqSd9jo7zceDfPRogM7bkkoNVv7'
+                        'DRNLEoB8HoirMgH969NrgL8jNzLEegqFzPRWM37GXd4uE8uuRkx4LAe',
+                    'public_key':
+                        'xpub661MyMwAqRbcGWtPvbWh9sc2BCfw2cTeVDYF23o3N1t6UZ5wv3EMm'
+                        'Dgp66FxHuDtWdft3B5eL5xQtyzAtkdmhhC95gjRjLzSTdkho95asu9',
+                    'address_generator': {
+                        'name': 'deterministic-chain',
+                        'receiving': {'gap': 17, 'maximum_uses_per_address': 3},
+                        'change': {'gap': 10, 'maximum_uses_per_address': 3}
+                    }
+                },
+                {
+                    'certificates': {'a': 'b'},
+                    'name': 'Account 2',
+                    'ledger': 'lbc_mainnet',
+                    'modified_on': 123,
+                    'seed':
+                        "carbon smart garage balance margin twelve chest sword toast envelope bottom stomac"
+                        "h absent",
+                    'encrypted': True,
+                    'private_key':
+                        'xprv9s21ZrQH143K42ovpZygnjfHdAqSd9jo7zceDfPRogM7bkkoNVv7'
+                        'DRNLEoB8HoirMgH969NrgL8jNzLEegqFzPRWM37GXd4uE8uuRkx4LAe',
+                    'public_key':
+                        'xpub661MyMwAqRbcGWtPvbWh9sc2BCfw2cTeVDYF23o3N1t6UZ5wv3EMm'
+                        'Dgp66FxHuDtWdft3B5eL5xQtyzAtkdmhhC95gjRjLzSTdkho95asu9',
+                    'address_generator': {
+                        'name': 'single-address',
+                    }
+                },
+            ]
+        }
+
+        storage = WalletStorage(default=wallet_dict)
+        wallet = Wallet.from_storage(storage, self.manager)
+        self.assertDictEqual(wallet_dict, wallet.to_dict())
+        jsonschema.validate(schema=self.wallet_schema, instance=wallet.to_dict())
```

Comment:
Could this just be added to one of the other tests which save and load the wallet, seems like it fits in well with the concept of testing saving/loading. I think the test right above this one would work?

If I missed something and this is doing something extra than what's in the existing saving/loading tests let me know.

Reply:
Yeah I actually started off adding this to the save/load test. I think the only thing that this is adding is testing both single and deterministic addresses (that's why there are two accounts). I figured it'd be simpler to have it be a separate test case rather than maintain a single test case with both purposes, but it's not a strong preference. Could I expand the test data for save/load test to include a second account?